R/contact_matrix.r

```diff
@@ -433,30 +431,26 @@ contact_matrix <- function(survey, countries = c(), survey.pop, age.limits, filt
 
   ## weights
   survey$participants[, weight := 1]
-  survey$contacts[, weight := 1]
 
   ## assign weights to participants, to account for weekend/weekday variation
   if (weigh.dayofweek) {
     found.dayofweek <- FALSE
-    for (table in surveys)
-    {
-      if ("dayofweek" %in% colnames(survey[[table]])) {
-        ## Add column sum_weight: Number of entries on weekdays / weekends
-        survey[[table]][, sum_weight := nrow(.SD),
-          by = (dayofweek %in% 1:5),
-        ]
+    if ("dayofweek" %in% colnames(survey$participants)) {
+      ## Add column sum_weight: Number of entries on weekdays / weekends
+      survey$participants[, sum_weight := nrow(.SD),
+        by = (dayofweek %in% 1:5),
+      ]
 
-        ## The sum of the weights on weekdays is 5
-        survey[[table]][dayofweek %in% 1:5, weight := 5 / sum_weight]
-        ## The sum of the weights on weekend is 2
-        survey[[table]][!(dayofweek %in% 1:5), weight := 2 / sum_weight]
+      ## The sum of the weights on weekdays is 5
+      survey$participants[dayofweek %in% 1:5, weight := 5 / sum_weight]
+      ## The sum of the weights on weekend is 2
+      survey$participants[!(dayofweek %in% 1:5), weight := 2 / sum_weight]
 
-        survey[[table]][, sum_weight := NULL]
-        found.dayofweek <- TRUE
+      survey$participants[, sum_weight := NULL]
+      found.dayofweek <- TRUE
 
-        # add boolean for "weekday"
-        survey[[table]][, is.weekday := dayofweek %in% 1:5]
-      }
+      # add boolean for "weekday"
+      survey$participants[, is.weekday := dayofweek %in% 1:5]
```

Comment:
For another PR (let's keep this one focused): it would be slightly clearer if we created the `is.weekday` column earlier and used it on L440, 445 & 447.